zipline/data/data_portal.py

```diff
@@ -606,20 +623,17 @@ def _get_history_daily_window_future(self, asset, days_for_window,
     def _get_market_minutes_for_day(self, end_date):
         return self.env.market_minutes_for_day(pd.Timestamp(end_date))
 
-    def _get_history_daily_window_equity(self, asset, days_for_window,
-                                         end_dt, field_to_use):
+    def _get_history_daily_window_equities(
+            self, assets, days_for_window, end_dt, field_to_use):
         ends_at_midnight = end_dt.hour == 0 and end_dt.minute == 0
 
-        # get the start and end dates for this sid
-        end_date = self._get_asset_end_date(asset)
-
-        if ends_at_midnight or (days_for_window[-1] > end_date):
```

Comment:
So far have not run into implications, the `load_raw_array` handles the case by providing `nans` before and after the window already.